packages/design-system/src/components/table/useTable/provider.tsx

```diff
@@ -121,6 +126,25 @@ export const TableProvider = ({
     [tableColumns]
   );
 
+  useEffect(() => {
+    if (selectedKey) {
+      const filteredRows = rows.filter((row) => {
+        if (!(row?.originalData as CookieTableData)?.parsedCookie) {
+          return true;
+        }
+
+        const tableCookieKey = getCookieKey(
+          (row?.originalData as CookieTableData)?.parsedCookie
+        );
+        return tableCookieKey === selectedKey;
+      });
+
+      if (filteredRows.length === 0) {
+        onRowClick(null);
+      }
+    }
+  }, [selectedKey, onRowClick, rows]);
+
```

Comment:
Need to add new optional callback prop for clearing row, can't put specific code inside table.

- A callback can be created in `CookieTable` component, which handles filtering callback.
- Use the callback, if passed as prop, unless always return true.

```suggestion
  useEffect(() => {
      const filteredRows = rows.filter((row) => isRowSelected?.(row.originalData) ?? true);

      if (filteredRows.length === 0) {
        onRowClick(null);
      }
  }, [onRowClick, rows]);

```

Reply:
Resolved in latest commit.